das/redis_storage_service.go

```diff
@@ -147,3 +147,11 @@ func (rs *RedisStorageService) ExpirationPolicy(ctx context.Context) ExpirationP
 func (rs *RedisStorageService) String() string {
 	return fmt.Sprintf("RedisStorageService(%+v)", rs.redisConfig)
 }
+
+func (rs *RedisStorageService) HealthCheck(ctx context.Context) error {
+	err := rs.client.Ping(ctx).Err()
+	if err != nil {
+		return err
+	}
+	return rs.baseStorageService.HealthCheck(ctx)
+}
```

Comment:
It's great that Redis has a ping built into it that we can use. This made me wonder if we want to have the health check fail fast if a cache layer fails but the underlying service is still usable, what do you think?

Reply:
Resolving based on offline discussion.